catalyst_voices/apps/voices/lib/widgets/document_builder/document_checkbox_builder_widget.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:catalyst_voices/widgets/document_builder/document_property_footer.dart';
+import 'package:catalyst_voices/widgets/document_builder/document_property_topbar.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:flutter/material.dart';
+
+class DocumentCheckboxBuilderWidget extends StatefulWidget {
+  final DocumentProperty property;
+  final ValueChanged<DocumentChange> onChanged;
+
+  const DocumentCheckboxBuilderWidget({
+    super.key,
+    required this.property,
+    required this.onChanged,
+  });
+
+  @override
+  State<DocumentCheckboxBuilderWidget> createState() =>
+      _DocumentCheckboxBuilderWidgetState();
+}
+
+class _DocumentCheckboxBuilderWidgetState
+    extends State<DocumentCheckboxBuilderWidget> {
+  bool editMode = false;
+  late bool value;
+  late bool initialValue;
+  DocumentNodeId get nodeId => widget.property.schema.nodeId;
+  String get description => widget.property.schema.description ?? '';
+  AgreementConfirmationDefinition get definition =>
+      widget.property.schema.definition as AgreementConfirmationDefinition;
+
+  @override
+  void initState() {
+    super.initState();
+
+    value = bool.tryParse(widget.property.value.toString()) ??
+        bool.tryParse(widget.property.schema.defaultValue.toString()) ??
+        false;
+    initialValue = value;
```

Comment:
This one is a bigger topic, in general there should be a way that a definition of the property also defines the type.

I haven't yet had time to do it but I'm thinking about adding a generic argument to the `BaseDocumentDefinition` which would define the type of `value` for a given property.

Then when you need to read the property value you could use the definition to process it for you. Do you want to do some experiments here?

Reply:
Maybe even the `DocumentProperty` could become `DocumentProperty<T>` and it would be filled up with already casted/parsed data instead of an Object. But imho the definion should do the casting, not the property itself

Reply:
In `DocumentDefinitionsObjectType` is `defaultValue` maybe we can do something similar to return a type base on the value?